startup/gui/menus.py

```diff
@@ -446,14 +447,20 @@ def __shaderNodeCreator( nodeName, shaderName ) :
 
 		return node
 
+	shaderPaths = os.environ["OSL_SHADER_PATHS"].split( os.path.pathsep )
+	# Remove 3Delight shaders from auto-discovery
+	if "DELIGHT" in os.environ :
+		for p in shaderPaths :
+			# Use `pathlib` to recognize the path despite Windows paths maybe having
+			# mixed forward and back slashes.
+			if str( pathlib.Path( os.environ["DELIGHT"] ) ) in str( pathlib.Path( p ) ) :
+				shaderPaths.remove( p )
```

Comment:
I wonder if we should/could do this by calling `ShaderUI.hideShaders()` from the 3Delight ShaderMenu code instead? Seems like it might be nice keep the related stuff together?

Reply:
That is nice to keep 3Delight handling a little more contained, done in https://github.com/GafferHQ/gaffer/pull/5459/commits/2ff04ae3607033a8c0b756e1b9a314d64f2bea65